pontoon/base/views.py

```diff
@@ -267,7 +267,10 @@ def translate(request, locale, slug, part=None, template='translate.html'):
             page = pages.get(name=part)
         except Subpage.DoesNotExist:
             # If page not specified or doesn't exist
-            page = pages.filter(resource__stats__locale=l)[0]
+            page = pages[0]
+            locale_pages = pages.filter(resource__stats__locale=l)
+            if locale_pages.exists():
```

Comment:
This does an extra query to the database outside of the one to fetch the locale pages, because you do one query for checking existence, then another to actually fetch the items.

`if locale_pages` will be True if at least one result exists, and it'll evaluate the queryset so that when you do `locale_pages[0]` below you use the cached results rather than doing another query.